File: University_Documentations/LearnPassVariableWithIntent/app/src/main/java/com/example/learnpassvariablewithintent/MainActivity.java
package com.example.learnpassvariablewithintent;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

import java.util.GregorianCalendar;

public class MainActivity extends AppCompatActivity {

    private Intent intent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        intent = new Intent(getApplicationContext(),ExtraActivity.class);



    }
    public void handleSubmit(View view){
        EditText editText = (EditText) findViewById(R.id.editTextMessage);

        Person person1 = new Person("Sahin","MARAL",new GregorianCalendar(2000,1,11));

        intent.putExtra("person1",person1);
        intent.putExtra("message",editText.getText().toString());
        startActivity(intent);
    }
}

File: DifferentActivities/app/src/main/java/com/example/differentactivities/MainActivity.kt
package com.example.differentactivities

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun changeActivity(view: View){
        val nameData = editText.text.toString()

        val intent = Intent(applicationContext,SecondActivity::class.java)
        intent.putExtra("name-from-mainActivity",nameData)
        startActivity(intent)
    }
}